### DISCUSSION TITLE
Problem displaying a raster data

### DISCUSSION THREAD

**Discussion by adlan2001** [DotSpatial/DotSpatial#1514] [Q&A]
I'm using the following code to open a raster file (specifically TIF file) and display it:
`Dim rasterSet As IRaster = Raster.Open(openRasterPath)`

Then this error came out when running the code:
`System.ApplicationException: 'The specified file type is not supported.'

Is there any other solution for this problem?

And, is it still relevant to use DotSpatial 1.9 since I'm depending on ChatGPT to code my simple GIS sofware app and it only can give the code referred to DotSpatial 1.9 (the latest version came after September 2021, which ChatGPT couldn't access it yet).
`

**Comment by MatthiasSchilder** [comment]
GDAL supports reading TIFFs. DotSpatial has an extension for using GDAL as a raster driver.

> And, is it still relevant to use DotSpatial 1.9 since I'm depending on ChatGPT to code my simple GIS sofware app and it only can give the code referred to DotSpatial 1.9 (the latest version came after September 2021, which ChatGPT couldn't access it yet).

Sorry, but this must be a joke. Learn how to program or hire a professional software developer. With your approach, you're building software you do not understand including vulnerabilities and issues the community overcame years ago.

**Reply by adlan2001** [reply]
I'm a student in a GIS course, and one of the assignments is to build a simple GIS software with simple tasks such as displaying data, measuring length and area, and simple query. I do have some understanding in other programming such as Python and C++, but VB.NET is something else for me. Took some times to understand it, try and error the codes suggested by ChatGPT. Anything else, I do agree with you about the vulnerabilities and issues of the older version and I'm still learning.

**Reply by MatthiasSchilder** [reply]
Do you know the DotSpatial DemoMap-application? It can do most things you've mentioned. Maybe a good starting point. You can find it in the repo. Besides, most DotSpatial code is written in C#, not VB.NET.
